src/model.c

```diff
@@ -95,155 +96,195 @@ int ir_model_time_lua(lua_State *L) {
+	lua_pushinteger(L, ALLEGRO_EVENT_TOUCH_BEGIN);
+	lua_setfield(L, -2, "EVENT_TOUCH_BEGIN");
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_MOUSE_WARPED);
-	lua_setfield(model->state, -2, "EVENT_MOUSE_WARPED");
+	lua_pushinteger(L, ALLEGRO_EVENT_TOUCH_END);
+	lua_setfield(L, -2, "EVENT_TOUCH_END");
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_TIMER);
-	lua_setfield(model->state, -2, "EVENT_TIMER");
+	lua_pushinteger(L, ALLEGRO_EVENT_TOUCH_MOVE);
+	lua_setfield(L, -2, "EVENT_TOUCH_MOVE");
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_DISPLAY_EXPOSE);
-	lua_setfield(model->state, -2, "EVENT_DISPLAY_EXPOSE");
+	lua_pushinteger(L, ALLEGRO_EVENT_TOUCH_CANCEL);
+	lua_setfield(L, -2, "EVENT_TOUCH_CANCEL");
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_DISPLAY_RESIZE);
-	lua_setfield(model->state, -2, "EVENT_DISPLAY_RESIZE");
+	lua_pushinteger(L, ALLEGRO_EVENT_DISPLAY_CONNECTED);
+	lua_setfield(L, -2, "EVENT_DISPLAY_CONNECTED");
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_DISPLAY_CLOSE);
-	lua_setfield(model->state, -2, "EVENT_DISPLAY_CLOSE");
+	lua_pushinteger(L, ALLEGRO_EVENT_DISPLAY_DISCONNECTED);
+	lua_setfield(L, -2, "EVENT_DISPLAY_DISCONNECTED");
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_DISPLAY_LOST);
-	lua_setfield(model->state, -2, "EVENT_DISPLAY_LOST");
+	lua_pushinteger(L, ALLEGRO_EVENT_DROP);
+	lua_setfield(L, -2, "EVENT_DROP");
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_DISPLAY_FOUND);
-	lua_setfield(model->state, -2, "EVENT_DISPLAY_FOUND");
+    // meta
+    lua_createtable(L, 0, 2);
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_DISPLAY_SWITCH_IN);
-	lua_setfield(model->state, -2, "EVENT_DISPLAY_SWITCH_IN");
+    // shader_meta
+	lua_createtable(L, 0, 1);
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_DISPLAY_SWITCH_OUT);
-	lua_setfield(model->state, -2, "EVENT_DISPLAY_SWITCH_OUT");
+	lua_pushcfunction(L, ir_shader_drop_lua);
+	lua_setfield(L, -2, "__gc");
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_DISPLAY_ORIENTATION);
-	lua_setfield(model->state, -2, "EVENT_DISPLAY_ORIENTATION");
+	lua_createtable(L, 0, 1);
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_DISPLAY_HALT_DRAWING);
-	lua_setfield(model->state, -2, "EVENT_DISPLAY_HALT_DRAWING");
+	lua_pushstring(L, "shader");
+	lua_setfield(L, -2, "__type");
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_DISPLAY_RESUME_DRAWING);
-	lua_setfield(model->state, -2, "EVENT_DISPLAY_RESUME_DRAWING");
+	lua_pushcfunction(L, ir_shader_use_lua);
+	lua_setfield(L, -2, "use");
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_TIMER);
-	lua_setfield(model->state, -2, "EVENT_TIMER");
+	lua_setfield(L, -2, "__index");
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_TOUCH_BEGIN);
-	lua_setfield(model->state, -2, "EVENT_TOUCH_BEGIN");
+    lua_setfield(L, -2, "shader");
+    // /shader_meta
 
-	lua_pushinteger(model->state, ALLEGRO_EVENT_TOUCH_END);
-	lua_setfield(model->state, -2, "EVENT_TOUCH_END");
+    // mat_meta
+	lua_createtable(L, 0, 3);
```

Comment:
Why not keep `ir` on the stack and have it be referenced twice? Probably not a big deal since this is one time initialization.